pkg/storage/unified/search/bleve.go

```diff
@@ -96,6 +100,37 @@ func (b *bleveBackend) GetIndex(ctx context.Context, key resource.NamespacedReso
 	return nil, nil
 }
 
+// updateIndexSizeMetric sets the total size of all file-based indices metric.
+func updateIndexSizeMetric(indexPath string, indexMetrics *resource.BleveIndexMetrics) {
+	if indexMetrics == nil {
+		return
+	}
+
+	for {
+		var totalSize int64
+
+		err := filepath.WalkDir(indexPath, func(path string, info os.DirEntry, err error) error {
+			if err != nil {
+				return err
+			}
+			if !info.IsDir() {
+				fileInfo, err := info.Info()
+				if err != nil {
+					return err
+				}
+				totalSize += fileInfo.Size()
+			}
+			return nil
+		})
+
+		if err == nil {
+			indexMetrics.IndexSize.Set(float64(totalSize))
+		}
+
+		time.Sleep(5 * time.Second)
```

Comment:
Given that typical scrape interval is often 10 or 15s, updating metrics more often than that seems unnecessary. Perhaps 1m interval would be enough here?